### PR TITLE
Added .eslintignore file | Suppress Eslint rules | Changed noopener to noreferrer

### PR DESCRIPTION
**Pull request by krishna8421**
- Added: 
.eslintignore
.husky/pre-commit
node_modules in .prettierignore

- Removed: 
.eslintrc

- Replaced:
```
"prettier/@typescript-eslint",
"plugin:prettier/recommended",
```
in .eslintrc.js with 
```
"prettier",
```

- Replaced
```
rel="noopener"
```
with 
```
rel="noreferrer"
```

- Separated some dev dependencies

- Changed 
class to className in [src/pages/donate/by-bitpay.tsx](https://github.com/xbmc/kodi-tv/blob/2413086356b090876e42e00ed995129eecefd4bd/src/pages/donate/by-bitpay.tsx#L31)

### CODE REVIEW

File: scripts/addon-parser/app.js [hunk 1]
```diff
@@ -107,7 +107,7 @@ let addonnodetype = "";
 let authornodetype = "";
 let categorynodetype = "";
 let data = "";
-
+let ah;
```
Comment (razzeee) [inline scripts/addon-parser/app.js:110]: This could be moved down two lines to be in the correct scope

**Comment by razzeee** [comment]
Not a fan of the husky changes, just tried it again and it still has the same problems. Running these checks on CI is far superior in my opinion.

**Comment by krishna8421** [comment]
Removed husky and changed the variable scope to the function

**Review by razzeee — APPROVED** [review]

**Comment by razzeee** [comment]
LGTM waiting for @pkscout

**Comment by pkscout** [comment]
I'm not going to approve any PR against main, especially ones that change the package-lock and package files.  Every time I have done that it ends up breaking the build at Netlify and I have to go and back everything out.  Please make a PR against staging, which means you should have the current staging code before making any changes.  The PR will run a test build in Netlify even before it is merged, so that will be a good first test.  Then when merged we get another build.  Once we're sure everything builds and works in staging, I can merge everything into main with pretty good assurance that it will build.  I have some other changes in staging already that I was going to push to production today, so if you can do the PR against staging today I will probably be able to merge it all to main yet today as all.

**Comment by krishna8421** [comment]
@pkscout Opened a [new PR](https://github.com/xbmc/kodi-tv/pull/235) for the staging branch.

**Pull request closed without merging**